verilog/analysis/checkers/struct_union_name_style_rule.cc

```diff
@@ -81,13 +83,33 @@ void StructUnionNameStyleRule::HandleSymbol(const verible::Symbol& symbol,
     }
     const auto* identifier_leaf = GetIdentifierFromTypeDeclaration(symbol);
     const auto name = ABSL_DIE_IF_NULL(identifier_leaf)->get().text();
-    if (!verible::IsLowerSnakeCaseWithDigits(name) ||
+    // get rid of the exceptions specified by user
+    std::string name_filtered;
+    if(raw_tokens.length()) {
+      name_filtered.reserve(name.length());
+      const auto &name_split = absl::StrSplit(name, '_');
+      const auto &exceptions = absl::StrSplit(raw_tokens, ',');
+      for(const auto &ns : name_split) {
+        if(std::find(exceptions.begin(), exceptions.end(), ns)
+            == exceptions.end()) {
+          name_filtered.append(ns);
+        }
+      }
+    }
+
+    if (!verible::IsLowerSnakeCaseWithDigits(name_filtered.size() ? name_filtered : name) ||
         !absl::EndsWith(name, "_t")) {
       violations_.insert(LintViolation(identifier_leaf->get(), msg, context));
     }
   }
 }
 
+absl::Status StructUnionNameStyleRule::Configure(absl::string_view configuration) {
+  using verible::config::SetString;
+  return verible::ParseNameValues(
+      configuration, {{"exceptions", SetString(&raw_tokens)}});
```

Comment:
so if `raw_tokens` would be a local variable, we can after `ParseNameValues()` str::Split() these into an array, so we don't have to do this every time, only at the time we configure.